nomad/job_endpoint.go

```diff
@@ -1412,7 +1412,7 @@ func (j *Job) List(args *structs.JobListRequest, reply *structs.JobListResponse)
 				paginator, err := paginator.NewPaginator(iter, tokenizer, filters, args.QueryOptions,
 					func(raw interface{}) error {
 						job := raw.(*structs.Job)
-						summary, err := state.JobSummaryByID(ws, namespace, job.ID)
+						summary, err := state.JobSummaryByID(ws, job.Namespace, job.ID)
```

Comment:
Can `summary` ever be `nil` in a working condition? Should we consider a `nil` summary an error as well?

Reply:
It looks like the state store function can return `nil, nil`, but we didn't check for that case previously which suggests it can't ever happen. Wouldn't be bad to add the check for belt-and-suspenders purposes though. Will do.